src/components/implementation/no_interface/llbooter/boot_deps.h

```diff
@@ -247,29 +251,23 @@ boot_done(void)
 static int
 boot_check_scheduler(char *comp_name)
 {
-	int i;
-	char *prefix = "sl_";
+	int ret;
+	const char prefix[] = "sl_";
 
-	for (i = 0; i < 3; i++) {
-		if (comp_name[i] == '\0' || comp_name[i] != prefix[i]) 
-			return 0;
-	}
+	ret = strncmp(comp_name, prefix, sizeof(prefix)-1);
 
-	return 1;
+	return !ret;
 }
 
 static int
 boot_check_shdmem(char *comp_name)
 {
-	int i;
-	char *prefix = "shmem";
+	int ret;
+	const char prefix[] = "shmem";
 
-	for (i = 0; i < 5; i++) {
-		if (comp_name[i] == '\0' || comp_name[i] != prefix[i])
-			return 0;
-	}
+	ret = strncmp(comp_name, prefix, sizeof(prefix)-1);
```

Comment:
This will cause friction for the convention is that `ret` is what we return.  The `!ret` will cause a double-take.

Likely should be:  `ret = !strncmp(comp_name, prefix, sizeof(prefix)-1);`